cpg_workflows/stages/rd_combiner.py

```diff
@@ -591,22 +642,41 @@ def queue_jobs(self, multicohort: MultiCohort, inputs: StageInput) -> StageOutpu
 @stage(required_stages=[AnnotateCohortSmallVariantsWithHailQuery])
 class SubsetMatrixTableToDatasetUsingHailQuery(DatasetStage):
     """
-    Subset the MT to a single dataset
+    Subset the MT to a single dataset - or a subset of families within a dataset
     Skips this stage if the MultiCohort has only one dataset
     """
 
     def expected_outputs(self, dataset: Dataset) -> dict[str, Path] | None:
         """
         Expected to generate a MatrixTable
-        This is kinda transient, so shove it in tmp
+        This is kinda transient, so shove it in tmp.
+
+        If subsetting to certain families, the output will be named accordingly
         """
         if len(get_multicohort().get_datasets()) == 1:
             get_logger().info(f'Skipping SubsetMatrixTableToDataset for single Dataset {dataset}')
             return None
-        return {
-            'mt': dataset.tmp_prefix() / 'mt' / self.name / f'{get_workflow().output_version}-{dataset.name}.mt',
-            'id_file': dataset.tmp_prefix() / f'{get_workflow().output_version}-{dataset.name}-SG-ids.txt',
-        }
+        if only_families := config_retrieve(['workflow', dataset.name, 'only_families'], []):
+            family_sg_data = query_for_family_sequencing_groups(
+                dataset.name,
+                config_retrieve(['workflow', 'sequencing_type']),
+                only_families,
+            )
```

Comment:
I think this use case should come before the one above, otherwise we'll never be able to apply this family-subset logic to single-cohort callsets like VCGS

That will mean we also need to apply the same decision downstream when deciding whether to use input from AnnotateCohort or the Subset step, even for single-project callsets